src/base/geometry.h

```diff
@@ -52,8 +53,11 @@ template <> struct from<Geometry::GeometryType>
             return Geometry::GeometryType::AngleType;
         else if (typeString == "distance")
             return Geometry::GeometryType::DistanceType;
-        else
+        else if (typeString == "torsion")
             return Geometry::GeometryType::TorsionType;
+        else
+            throw(std::runtime_error(
+                fmt::format("Unhandled geometry type '{}' - can't convert from TOML value.\n", std::string(typeString))));
```

Comment:
```suggestion
            throw toml::syntax_error(
                fmt::format("Unhandled geometry type '{}' - can't convert from TOML value.\n", std::string(typeString)), node.location());
```

Using the `toml::syntax_error` exception means that the error message will include the line number where the problem occurs.  Makes life much easier for the user debugging the input file.

Reply:
This is something I now realise from working on the (de)serialise for my new keyword type!